prog/phonons/initphonons.F90

```diff
@@ -0,0 +1,1612 @@
+  integer, allocatable, public :: iMovedAtoms(:)
+  integer, public :: nMovedAtom, nAtomUnitCell
+
+  !> Kpoints information
+  real(dp), allocatable, public :: kPoint(:,:), kWeight(:)
+  integer, public :: nKPoints
+  
+  !> maps atom index in central cell
+  integer, allocatable, public  :: Img2CentCell(:) 
+
+  !> neighbor list
+  type(TNeighbourList), public :: neighbourList
+
+  !> number of neighbors 
+  integer, allocatable, target, public :: nNeighbour(:)
+
+  !> cutoff for Hessian 
+  real(dp), public :: cutoff
+
+  !> Temperature range
+  real(dp), public :: TempMin, TempMax, TempStep
+
+  !> Modes to analyze (e.g., longitudinal, transverse, in-plane, etc)
+  integer, public :: selTypeModes
+
+  !> order=2 means harmonic, 3 is anharmonic 3rd order, etc. 
+  integer, public :: order
+
+  !> Atomic temperature
+  real(dp), public :: atTemperature
+
+  !> Whether modes should be computed
+  logical, public :: tCompModes
+
+  !> Whether modes should be plotted
+  logical, public :: tPlotModes
+
+  !> Whether modes should be animated
+  logical, public :: tAnimateModes
+
+  !> 
+  logical, public :: tXmakeMol
+
+  !>
+  logical, public :: tTransport
+
+  !> whether phonon dispersions should be computed
+  logical, public :: tPhonDispersion
+
+  !> number of repeated cells along lattice vectors 
+  integer, public :: nCells(3)
+
+  !> whether phonon dispersions should be computed
+  character(4), public :: outputUnits
+
+  !> whether taggedoutput should be written
+  logical, public :: tWriteTagged
+
+  !> Which phonon modes to animate
+  integer, allocatable, public :: modesToPlot(:)
+
+  !> Number of phonon modes to animate 
+  integer, public :: nModesToPlot
+
+  !> number of cycles in mode animation 
+  integer, public :: nCycles
+
+  !> number of steps in mode animation
+  integer, public, parameter :: nSteps = 10
+
+  !> Version of the current parser
+  integer, parameter :: parserVersion = 4
+
+  !> Version of the oldest parser, for which compatibility is maintained
+  integer, parameter :: minVersion = 4 
+
+  !> Container type for parser related flags.
+  type TParserFlags
+    logical :: tStop                        ! stop after parsing?
+    logical :: tIgnoreUnprocessed           ! Continue despite unprocessed nodes
+    logical :: tWriteHSD                    ! write parsed HSD input
+    logical :: tWriteTagged                 ! write TaggedOutput   
+  end type TParserFlags
+
+  !> constants parameters
+  integer, parameter, public :: ALLMODES = 0        ! along x,y,z  
+  integer, parameter, public :: XX = 1              ! along x 
+  integer, parameter, public :: YY = 2              ! along y 
+  integer, parameter, public :: ZZ = 3              ! along z 
+  integer, parameter, public :: LONGITUDINAL = 4    ! along z
+  integer, parameter, public :: TRANSVERSE = 5      ! along x,y
+  integer, parameter, public :: INPLANE = 6         ! along x,z
+  integer, parameter, public :: OUTOFPLANE = 7      ! along y
```

Comment:
Using the usual enumerator style is probably better for those, occupying `xx`, `yy` and `zz` could lead to problems later.

```suggestion
  type :: TPhononEnum
    !> along x,y,z  
    integer :: ALLMODES = 0
    !> along x 
    integer :: XX = 1
    !> along y 
    integer :: YY = 2
    !> along z 
    integer :: ZZ = 3
    !> along z
    integer :: LONGITUDINAL = 4
    !> along x,y
    integer :: TRANSVERSE = 5
    !> along x,z
    integer :: INPLANE = 6
    !> along y
    integer :: OUTOFPLANE = 7
  end type TPhononEnum

  !> Possible directions
  type(TPhononEnum), parameter, public :: phononEnum = TPhononEnum()
```